lib/validator.js

```diff
@@ -215,6 +244,25 @@ async function validateRawClientDataJson() {
 	return true;
 }
 
+async function validateTransports() {
+	// XXX: this isn't very useful, since this has already been parsed...
+	var transports = this.authnrData.get("transports");
+
+	if (transports != null && !Array.isArray(transports)) {
+		throw new Error("expected transports to be 'null' or 'array<string>'");
+	}
+
+	for (const index in transports) {
+		if (typeof transports[index] !== "string") {
+			throw new Error("expected transports[" + index + "] to be 'string'");
+		}
+	}
+
+	this.audit.journal.add("transports");
```

Comment:
Should this always be added, even if its null?

Reply:
Not mandatory, but yes. I've implemented the same behaviour used for userHandle.